### DISCUSSION TITLE
Access to other collection in a filter with a normal user

### DISCUSSION THREAD

**Discussion by indadev** [pocketbase/pocketbase#978] [Q&A] [answered]
Hi

I have 2 collections
  pages               -> list of page with data
  pages_relation -> a relation between pages

In trying to get the list of all subpage from a same parent page.

Something like this in the filter:
`filter: @collection.pages_relation.parent=${id} && id=@collection.pages_relation.page`

and Im getting this error:
`Uncaught ClientResponseError 403: Only admins can filter by @collection and @request query params.`

There is a way to do that filter with a normal user?
I know that I can do first a call to pages_relation to get all the pages id and later a call to pages to get the pages data, but with that Im going to make 2 calls, when I think I can only do one.

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
`@collection` and `@request` query filter params are not public for performance and security reason and they can be used only in the Collection API rules in the Admin UI.

If you are using PocketBase [v0.8.0-rc](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc2), we now have support for "Indirect expand", aka. expand where the relation is not in the main collection. Documentation will be available soon with the stable 0.8.0 release, but you could check the original issue for some info - https://github.com/pocketbase/pocketbase/issues/312#issuecomment-1242893496.

In your case, if I'm understanding your example correctly, that could look something like this:

```js
const rootPage = await pb.collection('pages').getOne('YOUR_ROOT_PAGE_ID', {
  'expand': 'pages_relation(parent).page',
});

// Example result:
{
    "id": "WyAw4bDrvws6gGl",
    "collectionId": "1rAwHJatkTNCUIN",
    "collectionName": "pages",
    "created": "2022-01-01 01:00:00.456Z",
    "updated": "2022-01-01 01:00:00.456Z",
    "title": "Lorem ipsum dolor sit...",
    "expand": {
        "pages_relation(parent)": [
            // subpage1 relation
            {
                "id": "lmPJt4Z9CkLW36z",
                "collectionId": "BHKW36mJl3ZPt6z",
                "collectionName": "pages_relation",
                "created": "2022-01-01 01:00:00.456Z",
                "updated": "2022-01-01 01:00:00.456Z",
                "parent": "WyAw4bDrvws6gGl",
                "page": "FtHAW9feB5rze7D",
                "expand": {
                    "page": {
                        "id": "FtHAW9feB5rze7D",
                        "collectionId": "1rAwHJatkTNCUIN",
                        "collectionName": "pages",
                        "created": "2022-01-01 00:00:00.000Z",
                        "updated": "2022-01-01 00:00:00.000Z",
                        "title": "Sub page 1 title..."
                    }
                }
            },
            // subpage2 relation ...
        ]
    }
}
```
The above will return the root page with all subpages related to it. You can apply the same `expand` expression when listing `pages`.

**Note that you client must have "View API rule" access in order to be able to expand the `pages_relation` and `pages_relation.page` collection records.**

**Reply by indadev** [reply]
Yeah thanks very much, that worked!!
Im really loving working with pockebase 👍
